### PR TITLE
chore: add tests for ReportPublicationGateway

### PR DESCRIPTION
**Pull request by krzysu**
PR #80 was merged, but it was missing the last thing, tests for ReportPublicationGateway.
I added them in this PR, but one test is not working as expected and I can't figure out why.
See comments below.

### CODE REVIEW

File: packages/react/src/publication/adapters/__tests__/ReportPublicationGateway.spec.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        publicationId,
+        reason: ReportReason.FAKE_ENGAGEMENT,
+        additionalComments,
+      });
+      const result = await gateway.report(request);
+
+      expect(result.isSuccess()).toBe(true);
+    });
+
+    it.skip(`should fail with AlreadyReportedError if publication was already reported`, async () => {
```
Comment (krzysu) [inline packages/react/src/publication/adapters/__tests__/ReportPublicationGateway.spec.ts:49]: notice that this test is skipped, as somehow the check for error instance is failing

File: packages/api-bindings/src/graphql/__helpers__/mutations.ts [hunk 2]
```diff
@@ -418,3 +421,29 @@ export function createCreateSetProfileMetadataViaDispatcherMutationMockedRespons
...
+export function createReportPublicationMutationWithErrorMockedResponse(args: {
+  variables: ReportPublicationMutationVariables;
+}): MockedResponse<ReportPublicationMutation> {
+  return {
+    request: {
+      query: ReportPublicationDocument,
+      variables: args.variables,
+    },
+    result: {
+      errors: [new GraphQLError('Publication already reported')],
```
Comment (krzysu) [inline packages/api-bindings/src/graphql/__helpers__/mutations.ts:446]: I think the mocked error response is correct..

Comment (krzysu) [reply packages/api-bindings/src/graphql/__helpers__/mutations.ts:446]: If I log the output of `console.log(e instanceof ApolloError, e);` in the `catch` part of `ReportPublicationGateway.ts` I'm getting:

```
false ApolloError: Publication already reported
        at new ApolloError (...)
   (...)
   {
      graphQLErrors: [ GraphQLError [Object] { message: 'Publication already reported' } ],
      clientErrors: [],
      networkError: null,
      extraInfo: undefined
    }
```
so it looks fine but the `instanceof` check is returning `false`

I don't know why

Comment (desfero) [reply packages/api-bindings/src/graphql/__helpers__/mutations.ts:446]: @krzysu PR that's fixing the issue https://github.com/lens-protocol/lens-sdk/pull/103

Comment (krzysu) [reply packages/api-bindings/src/graphql/__helpers__/mutations.ts:446]: I checked, and indeed that solves the issue, thx!

**Review by desfero — APPROVED** [review]

**Pull request merged**